src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/NonAssociationsResource.kt

```diff
@@ -527,6 +531,37 @@ class NonAssociationsResource(
     }
   }
 
+  @GetMapping("/non-associations/paged")
+  @PreAuthorize("hasRole('ROLE_READ_NON_ASSOCIATIONS')")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Get all non-associations, paged",
+    description = "Requires READ_NON_ASSOCIATIONS role",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "A page of Non-associations are returned",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the DELETE_NON_ASSOCIATIONS role with write scope.",
```

Comment:
I haven't looked at this yet but There is an argument for having this endpoint alongside the other migration/sync endpoints as this is mainly a synchronisation concern that "normal" users/clients wouldn't have (and it could confuse whoever is looking at the swagger docs)

Furthermore, wouldn't be best to require the same role as the other sync/migration role so that "normal" clients can't get all non-associations?

Reply:
Basically, I couldn't decide if it was just sync.  Its will continue to be used even after 2 way sync gets turned off so not sure it sits in that area.

Reply:
OK I'll move!

Reply:
my thinking is that a GET api for non-associations is quite a nice thing to have _anyway_. if we think more generically in a CRUD interface. we can add any old filters in futures that might be useful. a potential use case could be "gimme all the non-associations in Moorland"

Reply:
Fair enough. Just thinking aloud. I think Igor also pointed out there may be other use cases or clients that may be interest in getting all non-associations (for whatever reason). So maybe have that as part of the "normal" API it's OK.